jax/lax/lax.py

```diff
@@ -5121,7 +5142,7 @@ def _sort_batch_rule(batched_args, batch_dims, *, dimension, is_stable):
       new_args.append(batching.moveaxis(arg, bdim, new_bdim))
   new_dimension = dimension + (new_bdim <= dimension)
   bdims = (new_bdim,) * len(new_args)
-  return (sort_p.bind(*new_args, dimension=new_dimension, is_stable=is_stable),
+  return (sort_p.bind(*new_args, dimension=new_dimension, is_stable=is_stable, comparator=comparator),
```

Comment:
@jakevdp @SIben This change turns `sort_p` into a kind of higher-order primitive because it takes the comparator as a function. This is a serious increase in complexity of the primitive. Do we really want this? 

I am also wondering if the transformation rules for sort_p are correct for any comparator, or only for comparators with certain properties. I have not thought this carefully, but e.g., a comparator that works on unbatched data may not work on batched data. For sure, in jax2tf this change makes it quite a bit more complicated to convert sort_p.

I am wondering if it may be better to not parameterize sort_p by comparator, but instead parameterize it by `num_keys`, and perhaps also whether it is ascending or descending, but stop short of allowing any comparator function.

Reply:
Addressed in #3715